### PR TITLE
[azure] add azurestack-ipi-proxy workflow

### PR DESCRIPTION
**Pull request by jinyunma**
add workflow azurestack-ipi-proxy

### CODE REVIEW

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.13-installer-rehearse-azure-stack

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.13-installer-rehearse-azure-stack

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.13-installer-rehearse-azure-stack

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.13-installer-rehearse-azure-stack

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.13-installer-rehearse-azure-stack

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.13-installer-rehearse-azure-stack

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.13-installer-rehearse-azure-stack

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.13-installer-rehearse-azure-stack

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.13-installer-rehearse-azure-stack

**Comment by jinyunma** [comment]
cc @jianlinliu @gpei to review

File: ci-operator/step-registry/azure/deprovision/resourcegroup/azure-deprovision-resourcegroup-commands.sh [hunk 1]
```diff
@@ -10,16 +10,38 @@ az --version
...
 if [[ "${CLUSTER_TYPE}" == "azuremag" ]]; then
     az cloud set --name AzureUSGovernment
+elif [[ "${CLUSTER_TYPE}" == "azurestack" ]]; then
+    if [ ! -f "${CLUSTER_PROFILE_DIR}/cloud_name" ]; then
+        echo "Unable to get specific ASH cloud name!"
+        exit 1
+    fi
+    cloud_name=$(< "${CLUSTER_PROFILE_DIR}/cloud_name")
+
+    AZURE_AUTH_LOCATION="${CLUSTER_PROFILE_DIR}/osServicePrincipal.json"
```
Comment (jianlinliu) [inline ci-operator/step-registry/azure/deprovision/resourcegroup/azure-deprovision-resourcegroup-commands.sh:24]: The line is the same as the line 12, right?

Comment (jianlinliu) [inline ci-operator/step-registry/azure/deprovision/resourcegroup/azure-deprovision-resourcegroup-commands.sh:24]: The line is the same as the line 12, right?

File: ci-operator/step-registry/azure/deprovision/resourcegroup/azure-deprovision-resourcegroup-commands.sh [hunk 2]
```diff
@@ -10,16 +10,38 @@ az --version
...
+        --endpoint-resource-manager "${AZURESTACK_ENDPOINT}" \
+        --suffix-storage-endpoint "${SUFFIX_ENDPOINT}"
+    az cloud set --name ${cloud_name}
+    az cloud update --profile 2019-03-01-hybrid
 else
     az cloud set --name AzureCloud
 fi
+AZURE_AUTH_CLIENT_ID="$(<"${AZURE_AUTH_LOCATION}" jq -r .clientId)"
+AZURE_AUTH_CLIENT_SECRET="$(<"${AZURE_AUTH_LOCATION}" jq -r .clientSecret)"
+AZURE_AUTH_TENANT_ID="$(<"${AZURE_AUTH_LOCATION}" jq -r .tenantId)"
```
Comment (jianlinliu) [inline ci-operator/step-registry/azure/deprovision/resourcegroup/azure-deprovision-resourcegroup-commands.sh:44]: Is there any background to move the 3 lines here?

File: ci-operator/step-registry/azure/provision/vnet/azure-provision-vnet-commands.sh [hunk 3]
```diff
@@ -32,7 +32,9 @@ function create_disconnected_network() {
     local nsg rg="$1" subnet_nsgs="$2"
     for nsg in $subnet_nsgs; do
         run_command "az network nsg rule create -g ${rg} --nsg-name '${nsg}' -n 'DenyInternet' --priority 1010 --access Deny --source-port-ranges '*' --source-address-prefixes 'VirtualNetwork' --destination-address-prefixes 'Internet' --destination-port-ranges '*' --direction Outbound"
-        run_command "az network nsg rule create -g ${rg} --nsg-name '${nsg}' -n 'AllowAzureCloud' --priority 1009 --access Allow --source-port-ranges '*' --source-address-prefixes 'VirtualNetwork' --destination-address-prefixes 'AzureCloud' --destination-port-ranges '*' --direction Outbound"
+        if [[ "${CLUSTER_TYPE}" != "azurestack" ]]; then
```
Comment (jianlinliu) [inline ci-operator/step-registry/azure/provision/vnet/azure-provision-vnet-commands.sh:35]: Does that means ASH does nto need this nsg rule?

File: ci-operator/step-registry/azure/provision/vnet/azure-provision-vnet-commands.sh [hunk 4]
```diff
@@ -74,7 +97,11 @@ VNET_BASE_NAME="${NAMESPACE}-${JOB_NAME_HASH}"
 
 # create vnet
 vnet_arm_template_file="/tmp/01_vnet.json"
-run_command "curl -L https://raw.githubusercontent.com/openshift/installer/master/upi/azure/01_vnet.json -o ${vnet_arm_template_file}"
+vnet_template_url="https://raw.githubusercontent.com/openshift/installer/master/upi/azure/01_vnet.json"
```
Comment (jianlinliu) [inline ci-operator/step-registry/azure/provision/vnet/azure-provision-vnet-commands.sh:100]: I guess we can get the "01_vnet.json" at "/var/lib/openshift-install/upi/${CLUSTER_TYPE}/" from upi-installer:4.12 executor pod itself, not have to download it.

File: ci-operator/step-registry/azure/deprovision/resourcegroup/azure-deprovision-resourcegroup-commands.sh (hunk 1 above)
Comment (jinyunma) [reply ci-operator/step-registry/azure/deprovision/resourcegroup/azure-deprovision-resourcegroup-commands.sh:24]: sorry, should be `AZURE_AUTH_LOCATION="${SHARED_DIR}/osServicePrincipal.json"`.
On ASH, as service principal file for dev ASH PPE is not saved into `CLUSTER_PROFILE_DIR`, but mounted as secrets under namespace test-credentials([here](https://github.com/openshift/release/blob/master/ci-operator/step -registry/ipi/conf/azurestack/ipi-conf-azurestack-commands.sh#L28)), then move it into `${SHARED_DIR}`.

Do we need to consider PPE env when updating some azure provision step to support ASH? If not,line42~line44 can be moved back.

File: ci-operator/step-registry/azure/provision/vnet/azure-provision-vnet-commands.sh (hunk 3 above)
Comment (jinyunma) [reply ci-operator/step-registry/azure/provision/vnet/azure-provision-vnet-commands.sh:35]: This command failed to be executed on ASH. I checked there is no destination service tag named as "AzureCloud" on ASH WWT, it only exists on azure public cloud and azure gov cloud.
Seems no function impact without this rule on ASH, I checked that it is unreachable to connect from cluster to internet and cluster works well behind proxy.

File: ci-operator/step-registry/azure/deprovision/resourcegroup/azure-deprovision-resourcegroup-commands.sh (hunk 1 above)
Comment (jianlinliu) [reply ci-operator/step-registry/azure/deprovision/resourcegroup/azure-deprovision-resourcegroup-commands.sh]: Ah, got it. I think we can only take wwt ash into consideration for the steps only used in QE's wwt ash environment.

File: ci-operator/step-registry/azure/provision/vnet/azure-provision-vnet-commands.sh (hunk 3 above)
Comment (jianlinliu) [reply ci-operator/step-registry/azure/provision/vnet/azure-provision-vnet-commands.sh:35]: Got it.

File: ci-operator/step-registry/azure/deprovision/resourcegroup/azure-deprovision-resourcegroup-commands.sh (hunk 1 above)
Comment (jinyunma) [reply ci-operator/step-registry/azure/deprovision/resourcegroup/azure-deprovision-resourcegroup-commands.sh]: okay, will move line42~line44 back to original places

Comment (jinyunma) [reply ci-operator/step-registry/azure/deprovision/resourcegroup/azure-deprovision-resourcegroup-commands.sh]: updated

File: ci-operator/step-registry/azure/provision/bastionhost/azure-provision-bastionhost-commands.sh [hunk 5]
```diff
@@ -82,6 +82,27 @@ AZURE_AUTH_TENANT_ID="$(<"${AZURE_AUTH_LOCATION}" jq -r .tenantId)"
...
+        cp "${CLUSTER_PROFILE_DIR}/ca.pem" /tmp/ca.pem
+        cat /usr/lib64/az/lib/python*/site-packages/certifi/cacert.pem >> /tmp/ca.pem
+        export REQUESTS_CA_BUNDLE=/tmp/ca.pem
+    fi
+    az cloud register \
+        -n ${cloud_name} \
+        --endpoint-resource-manager "${AZURESTACK_ENDPOINT}" \
+        --suffix-storage-endpoint "${SUFFIX_ENDPOINT}"
+    az cloud set --name ${cloud_name}
+    az cloud update --profile 2019-03-01-hybrid
```
Comment (MayXuQQ) [inline ci-operator/step-registry/azure/provision/bastionhost/azure-provision-bastionhost-commands.sh:105]: get the profile_name from CLUSTER_PROFILE_DIR ?

Comment (jinyunma) [reply ci-operator/step-registry/azure/provision/bastionhost/azure-provision-bastionhost-commands.sh:105]: as dev PPE env also use same profile, it is hardcode in `ipi-conf-azurestack-creds`, to keep consistence with that, so also hardcode here.

**Pull request merged**